core/domain/wipeout_service.py

```diff
@@ -1553,14 +1564,24 @@ def _pseudonymize_models_transactional(
         datastore_services.put_multi(
             blog_post_models_list + blog_post_summary_models_list)
 
+        for model in blog_posts_related_models:
+            if isinstance(model, blog_author_details_model_class):
+                if model.author_id == user_id:
+                    model.author_id = pseudonymized_id
+                    model.update_timestamps()
+                    model.put()
+
     blog_post_ids_to_pids = (
         pending_deletion_request.pseudonymizable_entity_mappings[
             models.Names.BLOG.value])
-    for blog_post_id, pseudonymized_id in blog_post_ids_to_pids.items():
+    for blog_related_ids, pseudonymized_id in blog_post_ids_to_pids.items():
         blog_posts_related_models = [
             model for model in itertools.chain(
-                blog_post_models_list, blog_post_summary_models)
-            if model.id == blog_post_id
+                blog_post_models_list,
+                blog_post_summary_models,
+                [blog_author_details_model]
+                )
```

Comment:
Deindent by 4 to match opening indentation

Reply:
Done